app/code/BVMyBlog/Blog/Model/Blog.php

```diff
@@ -0,0 +1,155 @@
+<?php
+declare(strict_types=1);
+
+namespace BVMyBlog\Blog\Model;
+
+use BVMyBlog\Blog\Api\Data\BlogInterface;
+use BVMyBlog\Blog\Model\ResourceModel\Blog as ModelBlog;
+use Magento\Framework\Model\AbstractModel;
+
+/**
+ * Grid Model Blog
+ */
+class Blog extends AbstractModel implements BlogInterface
+{
+    /**
+     * Blog cache tag
+     */
+    const CACHE_TAG = 'bvmyblog_blog_post';
+
+    protected $_cacheTag = self::CACHE_TAG;
+
+    /**
+     * Prefix of model events names
+     *
+     * @var string
+     */
+    protected $_eventPrefix = 'bvmyblog_blog_post';
+
+    /**
+     * @inheritdoc
+     */
+    protected function _construct()
+    {
+        $this->_init(ModelBlog::class);
+    }
+
+    /**
+     * Function getIdentities
+     */
+    public function getIdentities()
+    {
+        return [self::CACHE_TAG . '_' . $this->getId()];
+    }
+
+    /**
+     * Retrieve post id
+     *
+     * @return int
+     */
+    public function getPostId() : int
+    {
+        $id = $this->_getData(self::POST_ID);
+
+        if ($id !== null) {
+            $id = (int)$id;
+        }
+
+        return $id;
```

Comment:
return $this->_getData(self::POST_ID) ? (int)$this->_getData(self::POST_ID) : null